rdflib/namespace.py

```diff
@@ -169,35 +181,38 @@ def __new__(cls, value):
         return rt
 
     def __mod__(self, *args, **kwargs):
-        return URIRef(str(self).__mod__(*args, **kwargs))
+        return URIRef(super().__mod__(*args, **kwargs))
 
     def format(self, *args, **kwargs):
-        return URIRef(str.format(self, *args, **kwargs))
+        return URIRef(super().format(*args, **kwargs))
 
     def __repr__(self):
-        return "URIPattern(%r)" % str(self)
+        return f"URIPattern({self!r})"
```

Comment:
@aucampia Is it [the same issue](https://github.com/RDFLib/rdflib/pull/1213/files#r633624221) here?

Reply:
yes, it also causes infinite recursion.